app/utils/dataHooks.ts

```diff
@@ -324,6 +326,28 @@ export function useCalcBakerStakeCooldownUntil() {
     );
 }
 
+/** Hook for calculating when a stake increase will take effect, will result in undefined while loading */
+export function useStakeIncreaseUntil() {
+    const status = useRewardBlockAndConsensusStatus();
+    const now = useCurrentTime(60000);
+
+    if (!status) {
+        return undefined;
+    }
+
+    const { bs, cs, rs } = status;
+
+    if (isBlockSummaryV1(bs)) {
+        if (!isRewardStatusV1(rs)) {
+            throw new Error('Block summary and reward status do not match.'); // Should not happen, as this indicates rs and bs are queried with different blocks.
+        }
+
+        return getV1Cooldown(0, bs, cs, rs.nextPaydayTime, now);
+    }
+
+    return getV0Cooldown(0, cs, now);
```

Comment:
Funny story. It is used, because the `PickBakerStakeAmount` is used by both the old and new flows. :clown_face: 
So I re-added the support for it (and gave it the correct epoch amounts)

Reply:
I also sneaked some fixes to headers in the old flows into the commit.